tsdb/chunkenc/histogram.go

```diff
@@ -283,39 +289,45 @@ func (a *HistogramAppender) appendable(h *histogram.Histogram) (
 	if value.IsStaleNaN(a.sum) {
 		// If the last sample was stale, then we can only accept stale
 		// samples in this chunk.
+		counterResetHint = UnknownCounterReset
 		return
 	}
 
 	if h.Count < a.cnt {
 		// There has been a counter reset.
-		counterReset = true
+		counterResetHint = CounterReset
 		return
 	}
 
 	if h.Schema != a.schema || h.ZeroThreshold != a.zThreshold {
+		// This case might or might not go along with a counter reset and that
```

Comment:
"that" appears spurious to me:

```suggestion
		// This case might or might not go along with a counter reset and
```